Refactor revision.go: move the reconciling logic from revision.go to cruds.go

ZhiminXiang opened the pull request:
Move the logic of reconciling Revision components (e.g. Deployment, Service, Autoscaler, etc.) from revision.go to cruds.go.

The main reasons of refactoring revision.go are:
1. revision.go becomes too crowded.
2. being consistent with the pattern of pkg/controller/route

tcnghia commented:
The `reconcileXXX` functions here have a lot more logic (branches/switches) than merely simple CRUDs operation.  I'd move them back to `revision.go` or may be a different file `resource_reconcile.go` to separate them from   other more straight-forward functions.

ZhiminXiang commented:
@tcnghia I made the following changes:
1. Moved functions reconcileXXX from revision/cruds.go to revision/reconcile_resources.go
2. Renamed file route/cruds.go to route/reconcile_resources.go since this file is mainly related to reconciling Route resources.

tcnghia commented:
/lgtm
/approve
